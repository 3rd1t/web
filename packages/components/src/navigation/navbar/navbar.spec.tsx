import React from "react"
import Navbar from "./navbar"
import { render } from "@testing-library/react"

describe(" Navbar", () => {
  it("renders correctly", () => {
    const { baseElement } = render(
      <Navbar
        logo={<span>I am a logo</span>}
        links={[
          {
            label: "Product",
            href: "#product",
          },
          {
            label: "Features",
            href: "#features",
          },
          {
            label: "Team",
            href: "#team",
          },
          {
            label: "Documentation",
            href: "#",
          },
        ]}
      />,
    )
    expect(baseElement).toMatchSnapshot()
  })
})
