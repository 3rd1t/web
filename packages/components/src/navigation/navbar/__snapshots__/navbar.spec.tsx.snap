// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Navbar renders correctly 1`] = `
<nav
  className="container flex flex-wrap items-center justify-between h-16 px-4 mx-auto md:flex-no-wrap"
  role="navigation"
>
  <a
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <span>
      I am a logo
    </span>
  </a>
  <div
    className="hidden md:block"
  >
    <div
      className="w-auto"
    >
      <ul
        className="flex items-center space-x-10"
      >
        <li>
          <a
            className="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
            href="#product"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Product
          </a>
        </li>
        <li>
          <a
            className="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
            href="#features"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Features
          </a>
        </li>
        <li>
          <a
            className="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
            href="#team"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Team
          </a>
        </li>
        <li>
          <a
            className="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Documentation
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="block md:hidden"
  >
    <div>
      <button
        aria-label="open menu"
        className="p-2 text-gray-700 hover:text-gray-900 focus:outline-none"
        data-testid="open menu"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          className="w-6 h-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M4 6h16M4 12h16M4 18h16"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
        </svg>
      </button>
    </div>
  </div>
</nav>
`;
